scripts/partition_manager.py

```diff
@@ -372,10 +372,26 @@ def _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynami
         if align_if_required(i, dynamic_partitions, False, reqs, solution):
             _set_addresses_and_align(reqs, sub_partitions, solution, size, start, dynamic_partitions)
 
+def align_is_valid(align):
+    if 'start' in align:
+        if isinstance(align['start'],list) and not (isinstance(x,int) for x in align['start']):
+            print(f"elements in {align['start']} is not type of {int}")
+            raise TypeError
+        elif not isinstance(align['start'], int):
+            print(f"{align['start']} is not type of {int}")
+            raise TypeError
+    if 'end' in align:
+        if isinstance(align['end'],list) and not (isinstance(x,int) for x in align['end']):
+            print(f"elements in {align['end']} is not type of {int}")
+            raise TypeError
+        elif not isinstance(align['end'], int):
+            print(f"{align['end']} is not type of {int}")
+            raise TypeError
```

Comment:
I think this will achieve the same result in fewer lines of code.

```suggestion
    alignment = align['start'] if 'start' in align else align['end']
    if (isinstance(alignment, list) and not (isinstance(x, int) for x in alignment)) or \
            (not isinstance(alignment, int)):
        raise TypeError(f"elements in {align['end']} is not type of {int}")
```